### ISSUE TITLE
Update ClusterBase AssociateWithGroup method

### ISSUE THREAD

**Issue by jepenven-silabs** [project-chip/connectedhomeip#11850]
#### Problem
Current implementation is tailor made for testing group communication.

 #### Proposed Solution
Refactor the Test part so that we don't need a `DeviceProxy` object in order to handle group methods (invoke commands/ write attributes)

**Comment by turon** [comment]
Reference to code in question: https://github.com/project-chip/connectedhomeip/blob/ddb7ce76a2459b7d3a2f9434dbf86c7e5b1a6a6d/examples/chip-tool/templates/partials/test_cluster.zapt#L270

It looks like both group and non-group cases are handled, so perhaps this can be closed:

```
 {{#if isGroupCommand}}
        cluster.AssociateWithGroup({{>device}}, groupId);
 {{else}}
        cluster.Associate({{>device}}, endpoint);
{{/if}}
```

Removing tag `v1_triage_split_4`

**Comment by jepenven-silabs** [comment]
@turon Wrong code quoted.

Problem is still present in master. The code assumes that the `DeviceProxy` object will have a valid SecureSession which is always true in the context of a `yaml` test but won't be in other context (e.g. inside Chip-tool ? ).

So this issue is still valid and still need to be fixed

```
CHIP_ERROR ClusterBase::AssociateWithGroup(DeviceProxy * device, GroupId groupId)
{
    // TODO Update this function to work in all possible conditions Issue #11850

    CHIP_ERROR err = CHIP_NO_ERROR;

    mDevice = device;
    if (mDevice->GetSecureSession().HasValue())
    {
        // Local copy to preserve original SessionHandle for future Unicast communication.
        Optional<SessionHandle> session = mDevice->GetExchangeManager()->GetSessionManager()->CreateGroupSession(
            groupId, mDevice->GetSecureSession().Value()->AsSecureSession()->GetFabricIndex());
        // Sanity check
        if (!session.HasValue() || !session.Value()->IsGroupSession())
        {
            err = CHIP_ERROR_INCORRECT_STATE;
        }

        mGroupSession.Grab(session.Value());
    }
    else
    {
        // something fishy is going on
        err = CHIP_ERROR_INCORRECT_STATE;
    }

    // Set to 0 for now.
    mEndpoint = 0;

    return err;
}
```

**Comment by jepenven-silabs** [comment]
This might need to be fixed before #1818

**Comment by jepenven-silabs** [comment]
Not needed after all. #1818 was done without this changes.
